### ISSUE TITLE
Review flutter attach performance

### ISSUE THREAD

**Issue by xster** [flutter/flutter#53313]
On a simple test, it takes ~8s to connect to an already running flutter observatory via `flutter attach`. Though ~3s was from cleaning up the devfs and closing the process. 

Log gist at https://gist.github.com/xster/5becc4937c2cef8d5e84660d0601abd1

There's a 2397ms apparently pure Dart part in the middle of the process too. 

We should optimize.

**Comment by jonahwilliams** [comment]
It's initializing the frontend server and compiling the application code. This process is bootstrapped from the already compiled app if available.

**Comment by xster** [comment]
Ah, that sounds reasonable. You're saying that middle 2s+ would be drastically reduced in profile mode right?

**Comment by jonahwilliams** [comment]
Yeah, under profile mode we wouldn't attempt to compile anything - just connect

**Comment by jonahwilliams** [comment]
there also shouldn't be any devfs work in profile mode - so if that is case there is definitely a bug

**Comment by zanderso** [comment]
@xster @jonahwilliams were either of you able to verify whether extra work was being done in profile mode?

**Comment by xster** [comment]
It was not. The purpose of `flutter attach` is to connect to a previously detached full-flutter app or to an add-to-app module in debug rather than profile mode.

**Comment by zanderso** [comment]
Is there anything actionable left in this issue, or should we close it?

**Comment by xster** [comment]
I think we should profile whether any of the 8s spent during the attach can be improved on. If not, we should provide clearer output on what the tool is doing while the user's waiting.

**Comment by jonahwilliams** [comment]
@xster would you be willing to try this again? I've landed a few optimizations to devFS initialization which should help. Preferably one that is uncached and one that is fully cached
